specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+There is no formal specification of the Kubernetes API and Resource Model. This
+document assumes Kubernetes 1.21 behavior; this behavior will typically be
+supported by many future Kubernetes versions. Additionally, this document may
+reference specific core Kubernetes resources; these references may be
+illustrative (i.e. _an implementation on Kubernetes_) or descriptive (i.e. _this
+Kubernetes resource MUST be exposed_). References to these core Kubernetes
+resources will be annotated as either illustrative or descriptive.
+
+This document considers two users of a given Knative Eventing environment, and
+is particularly concerned with the expectations of developers (and language and
+tooling developers, by extension) deploying applications to the environment.
+
+- **Developers** configure Knative resources to implement an event-routing
+  architecture.
+- **Operators** (also known as **platform providers**) provision the underlying
+  event routing resources and manage the software configuration of Knative
+  Eventing and the underlying abstractions.
+
+## RBAC Profile
+
+In order to validate the controls described in
+[Resource Overview](#resource-overview), the following Kubernetes RBAC profile
+may be applied in a Kubernetes cluster. This Kubernetes RBAC is an illustrative
+example of the minimal profile rather than a requirement. This Role should be
+sufficient to develop, deploy, and manage event routing for an application
+within a single namespace. Knative Conformance tests against "MUST", "MUST NOT",
+and "REQUIRED" conditions are expected to pass when using this profile:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: knative-developer
+rules:
+  - apiGroups: ["eventing.knative.dev"]
+    resources: ["broker", "trigger"]
+    verbs: ["get", "list", "create", "update", "patch", "delete"]
+  - apiGroups: ["messaging.knative.dev"]
+    resources: ["channel", "subscription"]
+    verbs: ["get", "list", "create", "update", "patch", "delete"]
+```
+
+In order to support resolving resources which meet the
+[Addressable](./overview.md#addressable) contract, the system controlling the
+[Trigger](#trigger-lifecycle) or [Subscription](#subscription-lifecycle) will
+need _read_ access to these resources. On Kubernetes, this is most easily
+achieved using role aggregation; on systems using Kubernetes RBAC, resources
+which wish to participate in Addressable resolution should provide the following
+`ClusterRole`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <component>-addressable-resolver
+  labels:
+    duck.knative.dev/addressable: "true"
+rules:
+  - apiGroups: [<resource apiGroup>]
+    resources: [<resource kind >, <resource kind>/status]
+    verbs: ["get", "list", "watch"]
+```
+
+This configuration advice SHALL NOT indicate a requirement for Kubernetes RBAC
```

Comment:
s/SHALL NOT/does not/
Or just remove the sentence since you already stated above that this is all informational.

Reply:
because it's not normative. This isn't something the implementer of the spec must adhere to or do anything about. It's just telling the reader that they don't have to implement Kube RBAC - that's just non-normative guidance, not a requirement. If anything it more like a requirement on the authors of this spec or the conformance test authors.

Reply:
Funny enough, I was going to comment about this entire section feeling a bit too prescriptive, until I read this sentence. Ignoring the "SHALL NOT" part of it, it helped me understand that this was just our way of saying "we expect this type of auth into your system for testing" but saying it in Kube technical terms instead of English. So it helped.